zokrates_core/src/r1cs.rs

```diff
@@ -244,20 +245,15 @@ fn r1cs_expression<T: Field>(
     }
 }
 
-/// Returns the index of `var` in the vector `variables` or adds `var`.
+/// Returns the index of `var` in `variables`, adding `var` with incremented index if it not yet exists.
 ///
 /// # Arguments
 ///
-/// * `variables` - A mutual vector that contains all existing variables. Not found variables will be added.
+/// * `variables` - A mutual map that maps all existing variables to their index.
 /// * `var` - Variable to be searched for.
-fn get_variable_idx(variables: &mut Vec<String>, var: &String) -> usize {
-    match variables.iter().position(|r| r == var) {
-        Some(x) => x,
-        None => {
-            variables.push(var.to_string());
-            variables.len() - 1
-        }
-    }
+fn get_variable_idx(variables: &mut HashMap<String, usize>, var: &String) -> usize {
```

Comment:
I know this was already the case before this change, but maybe we shouldn't call `get` something that has side effects

Reply:
Naming is hard ;-) https://stackoverflow.com/questions/16201065/propose-how-to-name-a-get-create-get-function is a related discussion./

How about `provide_variable_idx`?